spec/purely-editorial.html

```diff
@@ -0,0 +1,91 @@
+<emu-clause id="sec-purely-editorial">
+	<h1>Purely editorial changes</h1>
+  <p>These changes to unrelated parts of the ecma262 specification are editorial and don't affect the behavior of the modified algorithms. They are refactorings used to share existing algorithm steps with the new algorithms introduced by this proposal.</p>
+
+  <emu-clause id="sec-set-objects">
+    <h1>Set Objects</h1>
+
+    <emu-clause id="sec-set-constructor">
+      <h1>The Set Constructor</h1>
+
+      <emu-clause id="sec-set-iterable">
+        <h1>Set ( [ _iterable_ ] )</h1>
+        <p>When the `Set` function is called with optional argument _iterable_, the following steps are taken:</p>
+        <emu-alg>
+          1. If NewTarget is *undefined*, throw a *TypeError* exception.
+          1. Let _set_ be ? OrdinaryCreateFromConstructor(NewTarget, *"%Set.prototype%"*, &laquo; [[SetData]] &raquo;).
+          1. Set _set_.[[SetData]] to a new empty List.
+          1. If _iterable_ is either *undefined* or *null*, return _set_.
+          1. Let _adder_ be ? Get(_set_, *"add"*).
+          1. <del>If IsCallable(_adder_) is *false*, throw a *TypeError* exception.</del>
+          1. <del>Let _iteratorRecord_ be ? GetIterator(_iterable_).</del>
+          1. <del>Repeat,</del>
+            1. <del>Let _next_ be ? IteratorStep(_iteratorRecord_).</del>
+            1. <del>If _next_ is *false*, return _set_.</del>
+            1. <del>Let _nextValue_ be ? IteratorValue(_next_).</del>
+            1. <del>Let _status_ be Call(_adder_, _set_, &laquo; _nextValue_ &raquo;).</del>
+            1. <del>IfAbruptCloseIterator(_status_, _iteratorRecord_).</del>
+          1. <ins>Return ? AddValuesFromIterable(_set_, _iterable_, _adder_).</ins>
+        </emu-alg>
+
+        <emu-clause id="sec-add-values-from-iterable" type="abstract operation">
+          <h1>
+            <ins>AddValuesFromIterable</ins> (
```

Comment:
The code reuse would be around 50%, and this was already defined separately in the Set constructor. We can keep this as-is now, but since it's purely editorial we can discuss it in the final PR to ecma262.